other/cdi-jpa-testing/src/test/java/org/hibernate/demos/jpacditesting/AbstractCdiTest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * License: Apache License, Version 2.0
+ * See the LICENSE file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.demos.jpacditesting;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.se.SeContainerInitializer;
+
+import org.hibernate.demos.jpacditesting.support.TransactionalConnectionProvider;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.junit4.WeldInitiator;
+import org.jnp.server.NamingBeanImpl;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+
+import com.arjuna.ats.jta.utils.JNDIManager;
+
+public abstract class AbstractCdiTest {
+
+    private static NamingBeanImpl NAMING_BEAN;
+
+    @Rule
+    public WeldInitiator weld = WeldInitiator.from(((Weld) SeContainerInitializer.newInstance()))
```

Comment:
In order to avoid type cast here, you can simply use Weld API instead of CDI itself.
Just doing `WeldInitiator.from(new Weld())...` should do the trick.

Also note that by doing this, you are running with enabled discovery (your CP wll be scanned for beans) whereas commonly a synthetic archive is used to reduce bootstrap time. Not sure you intended that, so just pointing it out.

Reply:
Ha, didn't know it'd be as simple as calling `new Weld()` :-)

> you are running with enabled discovery...

Yes, I did this intentionally. Having to specify all the classes explicitly is one of the things I find a bit cumbersome when using Arquillian. Bootstrap time was perfectly fine when I ran the test. But it's definitely good to keep this in mind. Thanks!